src/tests/ftest/pool/target_query.yaml

```
@@ -0,0 +1,87 @@
+hosts:
+  test_servers:
+    - server-A
+    - server-B
+    - server-C
+    - server-D
+    - server-E
+  test_clients:
+    - client-A
+timeout: 300
+server_config:
+  engines_per_host: 2
+  name: daos_server
+  servers:
+    0:
+      pinned_numa_node: 0
+      nr_xs_helpers: 1
+      fabric_iface: ib0
+      fabric_iface_port: 31416
+      log_file: daos_server0.log
+      bdev_class: nvme
+      bdev_list: ["aaaa:aa:aa.a"]
+      scm_class: dcpm
+      scm_list: ["/dev/pmem0"]
+      scm_mount: /mnt/daos0
+      targets: 2
+    1:
+      pinned_numa_node: 1
+      nr_xs_helpers: 1
+      fabric_iface: ib1
+      fabric_iface_port: 31517
+      log_file: daos_server1.log
+      bdev_class: nvme
+      bdev_list: ["bbbb:bb:bb.b"]
+      scm_class: dcpm
+      scm_list: ["/dev/pmem1"]
+      scm_mount: /mnt/daos1
+      targets: 2
+pool:
+    mode: 146
+    name: daos_server
+    scm_size: 2G
+    nvme_size: 16G
+    control_method: dmg
+    pool_query_timeout: 30
+container:
+    type: POSIX
+    control_method: daos
+ior:
+  api: "DFS"
+  client_processes:
+    np: 1
+  dfs_destroy: False
+  iorflags:
+    flags: "-w -F -k -G 1 -vv"
+  test_file: /testFile
+  transfer_size: 1M
+  block_size: 1G
+  objectclass: !mux
+    SX:
+      dfs_oclass: "SX"
+      target_usage_count: 20
+      notes: "SX should write to all Targets"
+    RP_2G1:
+      dfs_oclass: "RP_2G1"
+      target_usage_count: 2
+      notes: "RP_2G1 is Group G1 so it should write to Two replica targets with in 1 group only."
+    RP_2GX:
+      dfs_oclass: "RP_2GX"
+      target_usage_count: 20
+      notes: "RP_2GX is Group GX and so it should write to all targets with Two Replicas."
+    EC_2P1G1:
+      dfs_oclass: "EC_2P1G1"
+      target_usage_count: 3
+      notes: "EC_2P1G1 is Group G1 with EC_2P1, so it should write to 2data+1parity (3 targets),
+              with in 1 group only."
+    EC_4P2GX:
+      dfs_oclass: "EC_4P2GX"
+      target_usage_count: 18
+      notes: "EC_4P2GX is Group GX with EC_4P2, so it should write to 4data+2parity (6 targets),
+              among max groups (In this case 3 so 6x3=18 targets)"
+    EC_8P2GX:
+      dfs_oclass: "EC_8P2GX"
+      target_usage_count: 20
+      notes: "EC_8P2GX is Group GX with EC_8P2, so it should write to 8data+2parity (10 targets),
+              among max groups (In this case 2 so 10x2 targets)"
+
```

This is a new check but will need to be fixed.
```suggestion
              among max groups (In this case 2 so 10x2 targets)"
```